### ISSUE TITLE
Update protocol generator to include definitions of performatives and fields

### ISSUE THREAD

**Issue by solarw** [valory-xyz/open-aea#522]
At the moment Message class generated does not have a good programmatic definition of performatives and fields relation

This relation check is done in code as
```
 Check correct contents
            actual_nb_of_contents = len(self._body) - DEFAULT_BODY_SIZE
            expected_nb_of_contents = 0
            if self.performative == SigningMessage.Performative.SIGN_TRANSACTION:
                expected_nb_of_contents = 2
                enforce(
                    isinstance(self.terms, CustomTerms),
                    "Invalid type for content 'terms'. Expected 'Terms'. Found '{}'.".format(
                        type(self.terms)
                    ),
                )
                enforce(
                    isinstance(self.raw_transaction, CustomRawTransaction),
                    "Invalid type for content 'raw_transaction'. Expected 'RawTransaction'. Found '{}'.".format(
                        type(self.raw_transaction)
                    ),
                )
            elif self.performative == SigningMessage.Performative.SIGN_MESSAGE:
                expected_nb_of_contents = 2
                enforce(
                    isinstance(self.terms, CustomTerms),
                    "Invalid type for content 'terms'. Expected 'Terms'. Found '{}'.".format(
                        type(self.terms)
                    ),
                )
                enforce(
                    isinstance(self.raw_message, CustomRawMessage),
                    "Invalid type for content 'raw_message'. Expected 'RawMessage'. Found '{}'.".format(
                        type(self.raw_message)
                    ),
                )
```

but can be replaced with something like
```
class SomeMessage:

PERFORMATIVES_FIELDS = {Performative1: (("field1", Field1Type), ("str_field", str))}
```
and all the checks can be done  using this definition.

it will give more flexibility, like check on message creation with list of missing fields in error
introspection features, like code can check how to construct message or provide list of expected fields.

also can help to build automated tests of message encoding/decoding

**Comment by DavidMinarsch** [comment]
Closing as won't fix.

The proposal is still structurally valid — the generator still emits per-performative `if self.performative == X: enforce(isinstance(...))` chains. `packages/valory/protocols/http/message.py` alone contains 8 of these guards today, identical to the pattern shown in the original report. Nothing has changed in `aea/protocols/generator/base.py` in this direction.

That said, on reflection the refactor is not worth doing:

- **Blast radius via hash churn.** Regenerating every protocol package in this repo and in `open-autonomy` would cascade through `packages.json` and force hash bumps on every agent and service that depends on any of them — dozens of downstream packages in lockstep, for a cosmetic change to generated code.
- **No functional or correctness benefit.** The existing checks are correct and produce the same errors. A declarative `PERFORMATIVES_FIELDS` map would be easier to read and introspect, but not more correct.
- **Generator changes are nontrivial.** `aea/protocols/generator/base.py` would need real surgery plus error-message compatibility work so existing tests keep passing.
- **LLM era dilutes the introspection argument.** "What fields does performative X expect?" is a trivial question to ask against the generated code directly — either by reading it or by pointing an LLM at it. A declarative map is a nice-to-have, not a capability unlock.

No reported pain in ~3 years, no correctness issue, high-churn rewrite for aesthetic and introspection wins that no concrete user is blocked on. If someone eventually needs programmatic `{performative: fields}` metadata, deriving it from the existing generated code (or at runtime via a helper) is a cheaper path than rewriting the generator.
